### PR TITLE
Leverage num input

### PR DESCRIPTION
**Pull request by midas-myth**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206513187657165

### CODE REVIEW

File: src/components/LeverageSlider/LeverageSlider.tsx [hunk 1]
```diff
@@ -42,12 +42,13 @@ export function LeverageSlider(p: Props) {
     return { marksLabel, keyValueMap, valueKeyMap };
   }, [finalMarks]);
 
-  const sliderValue = valueKeyMap[value ?? 0] ?? DEFAULT_LEVERAGE_KEY;
+  const sliderValue = valueKeyMap[value ?? 0] ?? lerpValue2SliderKey(keyValueMap, value ?? 0);
```
Comment (gmxer) [inline src/components/LeverageSlider/LeverageSlider.tsx:45]: not sure about `?? 0` should not it be `x2`? I think it's not the very first key

File: src/components/LeverageSlider/LeverageSlider.tsx [hunk 2]
```diff
@@ -127,3 +128,40 @@ function generateKeyValueMap(marks: number[]) {
 
   return { keyValueMap, valueKeyMap };
 }
+
+function lerpValue2SliderKey(keyValueMap: { [key: number]: number }, value: number): number {
+  const sortedValues = Object.values(keyValueMap);
```
Comment (gmxer) [inline src/components/LeverageSlider/LeverageSlider.tsx:133]: let's explicitly sort values, order may vary I think

File: src/domain/synthetics/trade/useTradeboxState.ts [hunk 3]
```diff
@@ -637,6 +673,24 @@ export function useTradeboxState(
...
+  //         return;
+  //       }
+
+  //       setLeverageOption(newValue);
+  //     } else {
+  //       setLeverageInputValue(leverageOption?.toString() ?? "");
+  //     }
+  //   },
+  //   [leverageFocusedInput, leverageInputValue, leverageOption, setLeverageOption]
+  // );
```
Comment (gmxer) [inline src/domain/synthetics/trade/useTradeboxState.ts:692]: still needed

File: src/components/LeverageSlider/LeverageSlider.tsx [hunk 4]
```diff
@@ -127,3 +128,40 @@ function generateKeyValueMap(marks: number[]) {
 
   return { keyValueMap, valueKeyMap };
 }
+
+function lerpValue2SliderKey(keyValueMap: { [key: number]: number }, value: number): number {
```
Comment (gmxer) [inline src/components/LeverageSlider/LeverageSlider.tsx:132]: lerp confuses here, it in fact doesn't matter what kind of value is passed here, single value can't be "lerped"

I'd call it valueToSliderKey / sliderValueToSliderKey

**Review by gmxer — CHANGES_REQUESTED** [review]

File: src/components/LeverageSlider/LeverageSlider.tsx (hunk 2 above)
Comment (midas-myth) [reply src/components/LeverageSlider/LeverageSlider.tsx]: done

File: src/components/LeverageSlider/LeverageSlider.tsx (hunk 4 above)
Comment (midas-myth) [reply src/components/LeverageSlider/LeverageSlider.tsx]: done

File: src/domain/synthetics/trade/useTradeboxState.ts (hunk 3 above)
Comment (midas-myth) [reply src/domain/synthetics/trade/useTradeboxState.ts]: done

File: src/components/LeverageSlider/LeverageSlider.tsx (hunk 1 above)
Comment (midas-myth) [reply src/components/LeverageSlider/LeverageSlider.tsx]: Agree, made `valueKeyMap[finalMarks[0]]`

**Review by gmxer — DISMISSED** [review]

**Pull request merged**